crypto/rand/randfile.c

```diff
@@ -76,43 +77,52 @@ int RAND_load_file(const char *file, long bytes)
 #ifndef OPENSSL_NO_POSIX_IO
     struct stat sb;
 #endif
-    int i, n, ret = 0;
-    FILE *in;
+    int ret = 0;
+    int in;
+    ssize_t got;
+    size_t need;
 
     if (bytes == 0)
         return 0;
 
-#ifndef OPENSSL_NO_POSIX_IO
-    if (stat(file, &sb) < 0 || !S_ISREG(sb.st_mode)) {
-        RANDerr(RAND_F_RAND_LOAD_FILE, RAND_R_NOT_A_REGULAR_FILE);
+    in = open(file, O_RDONLY);
```

Comment:
On VMS, you still need the slew of little strings, something that would be easy to encode like this:

```
#define open_extra_params
#ifdef OPENSSL_SYS_VMS
# undef open_extra_params
# define open_extra_params ,"shr=get,put,upd,del","ctx=bin,stm","rfm=stm","rat=none","mrs=0"
#endif
    in = open(file, O_RDONLY, 0 open_extra_params);
```